### ISSUE TITLE
Can't login with PCFDev

### ISSUE THREAD

**Issue by pilootchoum** [18F/cg-dashboard#510]
hi there,

Got an issue using the demo deck,

I'm running the server with go on a PCFDev instance,

I have access to the login page on http://localhost:9999 url, but then when filling the fileds,

I'm redirected to a white page with the oauthcallback url : http://localhost:9999/oauth2callback?code=i0NToh&state=state

I can see an error about certificate in the logs : 
Unable to get access token from code i0NToh error Post https://uaa.local.pcfdev.io/oauth/token: x509: certificate signed by unknown authority

However I tried to put the SelfSigned certificate of my pcfdev app in the manifest.yml of CGDeck, but it doesnt seem to change anything,

Has anyone already tried this type of configuration and solved this problem?

Regards 
Michael

**Comment by pilootchoum** [comment]
I tried to get it to work by modifying for now my env Variable to use http:// protocol instead of https,

But once on the dashboard, no organisation is showing, however my CF Instance does contain an organisation, I don't really know how to get some debug information about this...

**Comment by msecret** [comment]
@jcscottiii as you're the most familiar with the cloud.gov backend setup, any idea about this?

**Comment by pilootchoum** [comment]
There might be a misconfiguration on my side,

But I can access the CF api with another interface or with the cf cli command line and list the organisations and other components (spaces, app etc..)

So i'm wondering how to debug why the interface won't show anything on the dashboard.

**Comment by suprenant** [comment]
Thinking this could be an issue for Australia DTO/DTA.

**Comment by tessereth** [comment]
I ran into this pcfdev problem as well. I went far down the rabbit hole with this so I may as well explain what I found here.

So pcfdev uses an un-trusted self-signed certificate so the go server (correctly) rejects the connection. I tried changing to `http` rather than `https` but there was something in the oauth process that required a secure connection (I remember finding a `require_https: true` setting in `/var/vcap/jobs/uaa/config/uaa.yml` before I gave up on that approach).

Next I tried trusting the cert. I grabbed the certificate from the server (using `echo -n | openssl s_client -connect uaa.local.pcfdev.io:443 -tls1 | sed -ne '/-BEGIN CERTIFICATE-/,/-END CERTIFICATE-/p'`) and added it to my OS's trusted certs but the connection still failed. This was because the certificate was missing the `CA:TRUE` flag.

Next I had a look to see if the go oauth library had an equivalent of `--skip-ssl-validation` but I couldn't find one.

Finally, what I did to actually get it working was
* I created my own self-signed certificate - `openssl req -x509 -newkey rsa:4096 -keyout custom_key.pem -out custom_cert.pem -days 365 -nodes`
* added it to my system trusted certificates 
* copied the files to `/var/vcap/jobs/gorouter/config` of the pcfdev vm - `ssh vagrant@192.168.11.11`, password `vagrant`, your IP address may vary
* updated `/var/vcap/jobs/gorouter/config/gorouter.yml` to use to use the new keys
* restarted the router using `sudo /var/vcap/bosh/bin/monit restart gorouter`

So I'm not super happy with that solution. But it did work and I'd already learnt too much of cloud foundry internals by that point. I just can't run `cf dev destroy` or I'll have to do the whole thing again.
